playground/backend/internal/utils/system_utils_test.go

```diff
@@ -42,3 +45,86 @@ func TestGetFuncName(t *testing.T) {
 		})
 	}
 }
+
+func Test_checkNumOfTheParallelJobs(t *testing.T) {
+	baseFileFolder := "executable_files"
+	type args struct {
+		workingDir        string
+		numOfParallelJobs int
+	}
+	tests := []struct {
+		name        string
+		args        args
+		prepareFunc func()
+		want        bool
+	}{
+		{
+			// Test case with calling checkNumOfTheParallelJobs when there is no code processing folders.
+			// As a result, want to receive true
+			name: "there is no code processing folders",
+			args: args{
+				workingDir:        "",
+				numOfParallelJobs: 0,
+			},
+			prepareFunc: func() {},
+			want:        true,
+		},
+		{
+			// Test case with calling checkNumOfTheParallelJobs when there is one code processing folder.
+			// As a result, want to receive true
+			name: "there is one code processing folder",
+			args: args{
+				workingDir:        "",
+				numOfParallelJobs: 2,
+			},
+			prepareFunc: func() {
+				err := os.MkdirAll(filepath.Join(baseFileFolder, "1"), fs.ModePerm)
+				if err != nil {
+					panic(err)
+				}
+			},
+			want: true,
+		},
+		{
+			// Test case with calling checkNumOfTheParallelJobs when the number of the code processing folders is equals numOfParallelJobs.
+			// As a result, want to receive false
+			name: "there is one code processing folder",
+			args: args{
+				workingDir:        "",
+				numOfParallelJobs: 1,
+			},
+			prepareFunc: func() {
+				err := os.MkdirAll(filepath.Join(baseFileFolder, "1"), fs.ModePerm)
+				if err != nil {
+					panic(err)
+				}
+			},
+			want: false,
+		},
+		{
+			// Test case with calling checkNumOfTheParallelJobs when the number of the code processing folders is more than numOfParallelJobs.
+			// As a result, want to receive false
+			name: "there is one code processing folder",
```

Comment:
I see you have same names for 3 tests above, let's make them different

Reply:
Changed.